lib/bootloader_setup.pm

```diff
@@ -181,25 +181,46 @@ Boot the default kernel recovery mode (selected in the "Advanced options ..."):
     boot_grub_item(2, 2);
 
 =cut
+# private function
+sub grub_key_down {
+    if (get_var('S390_ZKVM')) {
+        type_line_svirt '$\'\\e\'[B', use_printf => 1;
+    }
+    else {
+        wait_screen_change { send_key 'down' };
+    }
+}
+
+# private function
+sub grub_key_enter {
+    if (get_var('S390_ZKVM')) {
+        type_line_svirt '';
+    }
+    else {
+        send_key 'ret';
+    }
+}
+
 sub boot_grub_item {
     my ($menu1, $menu2) = @_;
     $menu1 = 3 unless defined($menu1);
     $menu2 = 1 unless defined($menu2);
```

Comment:
maybe you wanna "cleanup" these both too:
```suggestion
    $menu1 //= 3;
    $menu2 //= 1;
```

Reply:
This one comes from older PR commit, but fixed.